src/test/java/core/basesyntax/service/FileServiceImplTest.java

```diff
@@ -0,0 +1,149 @@
+
+    @BeforeClass
+    public static void beforeClass() {
+        fileService = new FileServiceImpl();
+    }
+
+    @Test //(expected = RuntimeException.class)
+    public void readFromFileNullInputPath_NotOk() {
+        String path = null;
+
+        try {
+            fileService.readFromFile(path);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown if value is null");
+    }
+
+    @Test
+    public void readFromFileInputEmptyLinePath_NotOk() {
+        String path = "";
+
+        try {
+            fileService.readFromFile(path);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown if value is null");
+    }
+
+    @Test
+    public void readFromFileInputIncorrectLinePath_NotOk() {
+        String path = "-";
+
+        try {
+            fileService.readFromFile(path);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown if value is null");
+    }
+
+    @Test
+    public void readFromFile_Ok() {
+        List<String> expected = new ArrayList<>();
+        expected.add("b,banana,20");
+        expected.add("b,apple,100");
+        expected.add("s,banana,100");
+        expected.add("p,banana,13");
+        expected.add("r,apple,10");
+        expected.add("p,apple,20");
+        expected.add("p,banana,5");
+        expected.add("s,banana,50");
+
+        String path = "src/test/inputdatatest.csv";
+
+        List<String> actual = fileService.readFromFile(path);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void writeToFileNullInputDataLine_NotOk() {
+        String data = null;
+        String fileName = "testresult.csv";
+
+        try {
+            fileService.writeToFile(null, fileName);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown if value is null");
+    }
+
+    @Test
+    public void writeToFileEmptyInputDataLine_NotOk() {
+        String data = "";
+        String fileName = "testresult.csv";
+
+        try {
+            fileService.writeToFile(null, fileName);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown if value is null");
+    }
+
+    @Test
+    public void writeToFileNullInputFileName_NotOk() {
+        String data = "Roses are read, violets are blue";
+        String fileName = null;
+
+        try {
+            fileService.writeToFile(null, fileName);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown if value is null");
+    }
+
+    @Test
+    public void writeToFileEmptyLineInputFileName_NotOk() {
+        String data = "Roses are read, violets are blue";
+        String fileName = "";
+
+        try {
+            fileService.writeToFile(null, fileName);
```

Comment:
probably writeToFile(data, fileName) should be in all places

Reply:
> probably writeToFile(data, fileName) should be in all places

fixed)